my_enumerable.rb

```diff
@@ -0,0 +1,19 @@
+module MyEnumerable
+  def all?(*)
+    result = true
+    each { |number| result = false unless yield number }
+    result
+  end
+
+  def any?(*)
+    result = false
+    each { |number| result = true if yield number }
+    result
+  end
+
+  def filter(*)
+    result = []
+    each { |number| result << number if yield number }
+    result
+  end
+end
```

Comment:
- [ ] This implementation for the enumerable is not returning what we are looking for, this is not the logic we are expecting for the enumerable, you are declaring the result as a boolean and calling it later after yield. Take a look at the below implementation.

This is my own implementation, refer to this and refactor your own logic.

  ``

    def my_all?(parameter = nil)
    my_each do |x|
      if block_given?
        return false unless yield x
      elsif parameter.is_a?(Regexp)
        return false unless parameter.match?(x)
      elsif parameter.is_a?(Class)
        return false unless x.is_a?(parameter)
      elsif parameter.nil?
        return false unless x
      else
        return false unless x == parameter
      end
    end
    true
  end 
    ``

Reply:
Hi, thank you for the review but I have to say that I honestly do not understand what you mean because I get the expected results when I print (p) the result. Could you offer a better explanation as to what you mean by it does not produce the expected results?